src/components/frame/Extensions/pages/Extensions/ExtensionsListingPage/utils.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  type InstallableExtensionData,
+  type AnyExtensionData,
+  type ExtensionConfig,
+  type InstalledExtensionData,
+} from '~types/extensions.ts';
+
+export const getExtensionData = (
+  items: ExtensionConfig[],
+  availableExtensionsData: InstallableExtensionData[],
+  installedExtensionsData: InstalledExtensionData[],
+): AnyExtensionData[] =>
+  items.reduce<AnyExtensionData[]>((acc, item) => {
+    if (!item || !item.extensionId) {
+      return acc;
+    }
+
+    const { extensionId } = item;
+
+    const availableExtensionInfo = availableExtensionsData.find(
+      (extension) => extension.extensionId === extensionId,
+    );
+    const installedExtensionInfo = installedExtensionsData.find(
+      (extension) => extension.extensionId === extensionId,
+    );
+
+    if (availableExtensionInfo) {
+      acc.push(availableExtensionInfo);
+    } else if (installedExtensionInfo) {
+      acc.push(installedExtensionInfo);
+    }
+
+    return acc;
+  }, []);
```

Comment:
I would propose refactoring this into
```suggestion
  items
    .map(item => item?.extensionId)
    .filter(notMaybe)
    .reduce<AnyExtensionData[]>((acc, extensionId) => {

    let extensionData = availableExtensionsData.find(
      (extension) => extension.extensionId === extensionId,
    );
    
    if (!extensionData) {
      extensionData =  installedExtensionsData.find(
        (extension) => extension.extensionId === extensionId,
      );
    }

    if (extensionData) {
      acc.push(extensionData);
    } 
    
    return acc;
  }, []);
```